processing/src/test/java/org/apache/druid/segment/join/table/IndexedTableJoinMatcherTest.java

```diff
@@ -48,12 +54,51 @@
   {
     public static class MakeDimensionProcessorTest
     {
+      @Mock
+      private DimensionSelector dimensionSelector;
+
       private static final String KEY = "key";
 
       static {
         NullHandling.initializeForTests();
       }
 
+      @Test(expected = UnsupportedQueryException.class)
+      public void testMatchMultiValuedRowCardinalityUnknownShouldThrowException()
+      {
+        MockitoAnnotations.initMocks(this);
+        ArrayBasedIndexedInts row = new ArrayBasedIndexedInts(new int[]{2, 4, 6});
+        Mockito.doReturn(row).when(dimensionSelector).getRow();
+        Mockito.doReturn(DimensionDictionarySelector.CARDINALITY_UNKNOWN).when(dimensionSelector).getValueCardinality();
+
+        IndexedTableJoinMatcher.ConditionMatcherFactory conditionMatcherFactory =
+            new IndexedTableJoinMatcher.ConditionMatcherFactory(
+                ValueType.STRING,
+                IndexedTableJoinMatcherTest::createSingletonIntList
+            );
+        Supplier<IntIterator> dimensionProcessor = conditionMatcherFactory.makeDimensionProcessor(dimensionSelector, false);
+        // Test match should throw exception
+        dimensionProcessor.get();
```

Comment:
CI is complaining about an un-used return value 🤔 Same on line 99

Reply:
Done